wurst/event/DamageEvent.wurst

```diff
@@ -0,0 +1,437 @@
+/** 
+    This package provides a light-weight damage detection system with priority-ordered listeners. 
+
+    **IMPORTANT** 
+        This package **doesn't** provide protection from cyclic damage.
+        Damage from code **must** be declared by the user using setNextEventDamageFromCode() before dealing damage.
+        
+        if DETECT_NATIVE_ABILITIES is true:
+            > Spell damage is converted into negative damage values without this system, so do not use any other damage detection method.
+            > Spell reduction abilities like Runed bracers or Elunes Grace have to be coded.
+            > Life Drain doesn't heal the caster.
+            > Locust Swarm needs its Damage Return Factor changed from 0.75 to -0.75.
+            > Mana Shield abilities need their Damage Absorbed % reduced to 0.00 (and the damage absorbtion part has to be coded)
+            > Anti-Magic Shell (amount-based) has to be coded.
+
+
+    To listen to any damage instance firing:
+
+       addDamageListener() -> 
+            print(getEventDamageSource().getName()+" dealt "+getEventDamageAmount().toString()+" damage to "+getEventDamageTarget().getName()+".")
+
+    If the order of firing of the listeners is important, you can give a priority to the listener:
+
+        addDamageListener(2) -> 
+            print("This fires after.")
+        addDamageListener(0) -> 
+            print("This fires before.")
+
+    Each damage instance has a DamageType associated to it.
+    CODE **must** be declared by the user using DamageEvent.setNextDamageFromCode() before dealing damage.
+    ATTACK and SPELL are detected by the system (if DETECT_NATIVE_ABILITIES is false then native abilities will be detected as ATTACK)
+
+        addDamageListener() -> 
+            switch getEventDamageType() 
+                case ATTACK 
+                    print("This damage is from an auto-attack.")
```

Comment:
"auto-attack" kinda implies it wouldn't happen on user ordered attack

Reply:
What would be the better term then? "unit's attack"?